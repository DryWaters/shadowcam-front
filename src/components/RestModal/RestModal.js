import React from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";

class RestModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: true,
      restTimeLeft: this.props.restTime,
      restInterval: null
    };
  }

  componentDidMount() {
    const restInterval = setInterval(() => {
      if (this.state.restTimeLeft === 0) {
        clearInterval(this.state.restInterval);
        this.props.stopRest();
      } else {
        this.setState(prevState => {
          return {
            restTimeLeft: prevState.restTimeLeft - 1
          };
        });
      }
    }, 1000);
    this.setState({
      restInterval
    });
  }

  toggle = () => {
    clearInterval(this.state.restInterval);
  };

  render() {
    return (
      <div>
        <Modal isOpen={this.state.modal} toggle={this.props.stopRest}>
          <ModalHeader toggle={this.toggle}>Take a rest!</ModalHeader>
          <ModalBody>
            <div>You deserve a break!</div>
            <div>
              Take a breather for another {this.state.restTimeLeft} seconds
            </div>
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={this.props.stopRest}>
              Cancel Break
            </Button>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

export default RestModal;
