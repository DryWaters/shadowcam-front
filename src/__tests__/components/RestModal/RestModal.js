import React from "react";
import { shallow } from "enzyme";
import { Button, Modal } from "reactstrap";
import RestModal from "../../../components/RestModal/RestModal";

describe("RestModal render", () => {
  it("Should render RestModal component without crashing", () => {
    shallow(<RestModal />);
  });

  it("Should match RestModal component snapshot", () => {
    const wrapper = shallow(<RestModal />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("RestModal toggle triggered", () => {
  it("Should call the stop prop function on click", () => {
    let spy = jest.fn();
    const wrapper = shallow(<RestModal stopRest={spy} />);
    wrapper.find(Button).simulate("click");
    expect(spy).toHaveBeenCalledTimes(1);
  });
});

describe("RestModal should be displayed as soon as started", () => {
  it("Should be displayed as soon as it is open", () => {
    const wrapper = shallow(<RestModal />);
    expect(wrapper.find(Modal).prop("isOpen")).toBe(true);
  });
});
