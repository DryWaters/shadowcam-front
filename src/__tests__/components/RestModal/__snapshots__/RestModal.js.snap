// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestModal render Should match RestModal component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
          Object {
            "mountOnEnter": true,
            "timeout": 150,
          }
        }
        centered={false}
        fade={true}
        isOpen={true}
        keyboard={true}
        modalTransition={
          Object {
            "timeout": 300,
          }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        role="dialog"
        zIndex={1050}
      >
        <ModalHeader
          charCode={215}
          closeAriaLabel="Close"
          tag="h5"
          wrapTag="div"
        >
          Take a rest!
        </ModalHeader>
        <ModalBody
          tag="div"
        >
          <div>
            You deserve a break!
          </div>
          <div>
            Take a breather for another 
             seconds!
          </div>
        </ModalBody>
        <ModalFooter
          tag="div"
        >
          <Button
            color="primary"
            tag="button"
          >
            Cancel Break
          </Button>
        </ModalFooter>
      </Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            tag="h5"
            wrapTag="div"
          >
            Take a rest!
          </ModalHeader>,
          <ModalBody
            tag="div"
          >
            <div>
              You deserve a break!
            </div>
            <div>
              Take a breather for another 
               seconds!
            </div>
          </ModalBody>,
          <ModalFooter
            tag="div"
          >
            <Button
              color="primary"
              tag="button"
            >
              Cancel Break
            </Button>
          </ModalFooter>,
        ],
        "fade": true,
        "isOpen": true,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "role": "dialog",
        "toggle": undefined,
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "charCode": 215,
            "children": "Take a rest!",
            "closeAriaLabel": "Close",
            "tag": "h5",
            "wrapTag": "div",
          },
          "ref": null,
          "rendered": "Take a rest!",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div>
                You deserve a break!
              </div>,
              <div>
                Take a breather for another 
                 seconds!
              </div>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You deserve a break!",
              },
              "ref": null,
              "rendered": "You deserve a break!",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Take a breather for another ",
                  undefined,
                  " seconds!",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Take a breather for another ",
                undefined,
                " seconds!",
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              color="primary"
              tag="button"
            >
              Cancel Break
            </Button>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel Break",
              "color": "primary",
              "onClick": undefined,
              "tag": "button",
            },
            "ref": null,
            "rendered": "Cancel Break",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={false}
          fade={true}
          isOpen={true}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          zIndex={1050}
        >
          <ModalHeader
            charCode={215}
            closeAriaLabel="Close"
            tag="h5"
            wrapTag="div"
          >
            Take a rest!
          </ModalHeader>
          <ModalBody
            tag="div"
          >
            <div>
              You deserve a break!
            </div>
            <div>
              Take a breather for another 
               seconds!
            </div>
          </ModalBody>
          <ModalFooter
            tag="div"
          >
            <Button
              color="primary"
              tag="button"
            >
              Cancel Break
            </Button>
          </ModalFooter>
        </Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              closeAriaLabel="Close"
              tag="h5"
              wrapTag="div"
            >
              Take a rest!
            </ModalHeader>,
            <ModalBody
              tag="div"
            >
              <div>
                You deserve a break!
              </div>
              <div>
                Take a breather for another 
                 seconds!
              </div>
            </ModalBody>,
            <ModalFooter
              tag="div"
            >
              <Button
                color="primary"
                tag="button"
              >
                Cancel Break
              </Button>
            </ModalFooter>,
          ],
          "fade": true,
          "isOpen": true,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "role": "dialog",
          "toggle": undefined,
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": "Take a rest!",
              "closeAriaLabel": "Close",
              "tag": "h5",
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": "Take a rest!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div>
                  You deserve a break!
                </div>,
                <div>
                  Take a breather for another 
                   seconds!
                </div>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "You deserve a break!",
                },
                "ref": null,
                "rendered": "You deserve a break!",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Take a breather for another ",
                    undefined,
                    " seconds!",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Take a breather for another ",
                  undefined,
                  " seconds!",
                ],
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                color="primary"
                tag="button"
              >
                Cancel Break
              </Button>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel Break",
                "color": "primary",
                "onClick": undefined,
                "tag": "button",
              },
              "ref": null,
              "rendered": "Cancel Break",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
